src/Container/Container.php

```diff
@@ -165,4 +222,44 @@ public function __call($name, $arguments)
             throw new \BadMethodCallException("No application method named {$name}()");
         }
     }
+
+    /**
+     * Instantiate an object of named class using reflection to recursively to resolve dependencies
+     *
+     * @param string $className
+     * @return mixed
+     * @throws \ReflectionException
+     */
+    protected function resolveInstance(string $className) {
+        $class = new \ReflectionClass($className);
+
+        if (!$class->isInstantiable()) {
+            throw new \ReflectionException("$className cannot be instantiated");
```

Comment:
maybe good to follow how the other exceptions are created,e.g.

`throw new \ReflectionException(sprintf('Class "%s" cannot be instantiated', $className));`